replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/data/AbstractPersistable.java

```diff
@@ -78,5 +94,6 @@ public void setVersion(int version) {
   public void fromMap(Map<String, Object> properties) {
     setId((String) properties.get(ID_KEY));
     setVersion((int) properties.get(VERSION_KEY));
+    this.modified = (Date) properties.get(MODIFIED_KEY);
```

Comment:
✏️ For backward compatibility, I would check if the map doesn't contain the modified timestamp and initialize it for the first time here such that it never be `null`